### DISCUSSION TITLE
HOW TO INSTALL ON MACOS??????

### DISCUSSION THREAD

**Discussion by SLIM3N3M** [philippe44/AirConnect#546] [General]
Im sorry im a noob and a beginner but this will littarally change my life as I have 17 GH/Nest speakers around me house!!! please I can't understand getup. I found some files but idk how to install them. It's been 2 weeks since I started trying. as I a developer I know you want this to be easy for us noobs but I really really need your help!

**Comment by philippe44** [comment]
I don't I understand what challenges you are facing but macOS as a platform is a pain. If you have a Pi handy, I'd recommend you try that first. Otherwise, have a good look at the REASME and post logs.

**Reply by SLIM3N3M** [reply]
<img width="3008" alt="Screenshot 2024-09-22 at 11 12 32 AM" src="https://github.com/user-attachments/assets/e0e70bbe-de6e-4b6c-915f-50a79607877f">

**Reply by SLIM3N3M** [reply]
I have reached here as of now... just getting here took me 2 weeks, What should I do next? no airplay devices have yet showed up in my airplay menu

**Reply by philippe44** [reply]
You are trying to use one of the packaged version made by another user. You should ask for help from him as he will be able to guide you better how to access the container he created.

**Reply by SLIM3N3M** [reply]
<img width="3008" alt="Screenshot 2024-09-22 at 2 21 22 PM" src="https://github.com/user-attachments/assets/5fce68f7-600e-4629-8cf8-e8488f19b4d3">
This I what I want to install and I want to install it your way!, the only problem I I don't understand!!! please help me bro! Im sorry I didn't know, I installed that image on docker by mistake! I have no other option but you bro, my last hope is you. my family has 15 GH speakers around the house and im having a mad headache trying to figure this out every damn night as a normal user. Please help me!, just teach me how to do it your way and you will change the lives in my house bro!

**Reply by SLIM3N3M** [reply]
![IMG_722E4F3D79A4-1](https://github.com/user-attachments/assets/59d02777-9ae9-48e3-9a66-637eac5c54f0)
I have also enabled UPNP in router but still no luck
<img width="3008" alt="Screenshot 2024-09-22 at 3 09 14 PM" src="https://github.com/user-attachments/assets/94d20230-1d88-4dbb-90c3-6357c0d605d6">

**Reply by philippe44** [reply]
With my version, there is not installation to be done, per README, you just download the zip file you find in the "tags" section on github, unzip and launch it. But macOS might not be the easiest is to do such "manual" processes and you might not be very familiar with computers. I suggest you go to one of the containarized versions and seek help from there.

**Reply by SLIM3N3M** [reply]
I would but no one else is replying, it's like no one but you actually cares about AirConnect!, I have been trying for days. if you are familiar with a contained version could you guide me? I have attached a screenshot of my logs above!, I agree im not really familiar with terminal but through basic step by step instructions I was able to get home bridge and all its plugins running. I am willing to learn how to do the "manual processes". if you could teach me!

**Reply by pwt** [reply]
Currently, airupnp and aircast are binding to the Docker internal network, not your home network, which is why no devices are being discovered.

You need **host networking** enabled in Docker settings (see screenshot), and the container must be started with the `--net=host` option.

![screenshot_1046](https://github.com/user-attachments/assets/2a02e753-7c33-4f9a-8bf1-1fa69cd875fd)

**Reply by SLIM3N3M** [reply]
PWD, you gave me hope in humanity and GitHub just by replying! may god bless you! So I followed your instructions and enabled host networking! attaching my logs for you to see, I have one question how do I check if the container has started with the --net=host option.

<img width="3008" alt="Screenshot 2024-09-22 at 4 09 03 PM" src="https://github.com/user-attachments/assets/53fb96a9-3752-4dc1-9f6f-9ec8ae2a82e7">

<img width="3008" alt="Screenshot 2024-09-22 at 4 08 37 PM" src="https://github.com/user-attachments/assets/28cc5934-1e0e-4172-8a76-ab869124f7ad">

**Comment by SLIM3N3M** [comment]
After searching for a bit longer I might have some new info, the IP address shown in the logs here is not my IP address even though host networking is on! any ideas?
<img width="1011" alt="Screenshot 2024-09-22 at 8 17 49 PM" src="https://github.com/user-attachments/assets/eff170be-19b0-4a1b-93dc-7b851d187456">

**Comment by pwt** [comment]
For some reason, on macOS and Docker with host networking enabled, containers started with `--net=host` don't bind to the host interface -- unlike on other platforms. I've tried forcing the bind:
```
docker run -d --net=host --rm --name=airconnect -e AIRUPNP_VAR="-b 192.168.0.106" 1activegeek/airconnect
```
but that throws a different error:
```
[17:04:23.927] Start:1094 Binding to iface 192.168.0.106:0 [(null)]
[17:04:23.930] bind_host:721 cannot bind: Cannot assign requested address
```
Hence, I'm not sure how to get AirConnect running on macOS in a container.

The native macOS binaries work fine, however. What model of Mac do you have (Intel or Apple Silicon)?

**Reply by SLIM3N3M** [reply]
> For some reason, on macOS and Docker with host networking enabled, containers started with `--net=host` don't bind to the host interface -- unlike on other platforms. I've tried forcing the bind:
> 
> ```
> docker run -d --net=host --rm --name=airconnect -e AIRUPNP_VAR="-b 192.168.0.106" 1activegeek/airconnect
> ```
> 
> but that throws a different error:
> 
> ```
> [17:04:23.927] Start:1094 Binding to iface 192.168.0.106:0 [(null)]
> [17:04:23.930] bind_host:721 cannot bind: Cannot assign requested address
> ```
> 
> Hence, I'm not sure how to get AirConnect running on macOS in a container.
> 
> The native macOS binaries work fine, however. What model of Mac do you have (Intel or Apple Silicon)?

I have a MacBook Air m1 docked next to my router running as a server for home bridge, I will delete docker and throw it in a trash can right now if you could tell me how to just do a basic install!!

**Comment by philippe44** [comment]
> For some reason, on macOS and Docker with host networking enabled, containers started with `--net=host` don't bind to the host interface -- unlike on other platforms. I've tried forcing the bind:
> 
Thanks for reminding me why I so much dislike macOS 😀.

**Reply by SLIM3N3M** [reply]
hahahaha I guess it is very annoying! on top of that you have to deal with idiots like me!, sorry Philippe I love what you did for us!
I would rather give you 150 bucks for helping us than to apple for 1 HomePod!! and I will too!
